File: unisync_android/app/src/main/java/com/anhquan/unisync/core/plugins/ring_phone/RingPhonePlugin.kt
package com.anhquan.unisync.core.plugins.ring_phone

import android.content.Intent
import com.anhquan.unisync.core.Device
import com.anhquan.unisync.core.DeviceConnection
import com.anhquan.unisync.core.plugins.UnisyncPlugin
import com.anhquan.unisync.models.DeviceMessage

class RingPhonePlugin(
    private val device: Device,
) : UnisyncPlugin(device, DeviceMessage.Type.RING_PHONE) {
    private object Method {
        const val RING = "ring"
    }

    override fun listen(
        header: DeviceMessage.DeviceMessageHeader,
        data: Map<String, Any?>,
        payload: DeviceConnection.Payload?
    ) {
        super.listen(header, data, payload)
        context.startForegroundService(Intent(context, RingPhoneService::class.java))
    }
}

File: lib/core/plugins/ring_phone/ring_phone.plugin.dart
import 'package:unisync/core/device.dart';
import 'package:unisync/core/plugins/base_plugin.dart';
import 'package:unisync/models/device_message/device_message.model.dart';

class RingPhonePlugin extends UnisyncPlugin {
  RingPhonePlugin(Device device)
      : super(device, type: DeviceMessage.Type.RING_PHONE);
  static const _Method = (RING: 'ring',);

  void ringMyPhone() {
    sendRequest(_Method.RING);
  }
}
